analyses/focal-cn-file-preparation/05-define-most-focal-cn-units.Rmd

```diff
@@ -0,0 +1,272 @@
+    # threshold
+    mutate(
+      dominant_arm_status = case_when(
+        callable_fraction < (1 - uncallable_threshold) ~ "uncallable",
+        loss_fraction > status_threshold ~ "loss",
+        gain_fraction > status_threshold ~ "gain",
+        loss_fraction + gain_fraction > status_threshold ~ "unstable",
+        TRUE ~ "neutral"
+      )
+    )
+
+# Display table
+consensus_seg_arm_status
+```
+
+### Determine cytoband status
+
+We want to include cytoband and gene-level calls for chromosome arms that have not been defined as a gain or loss to make the cytoband-level majority calls.
+
+```{r}
+# Now count the cytoband level calls (for each status call) and define
+# the cytoband as that status if more than the `status_threshold` fraction
+# value of the total counts are for that particular status
+consensus_seg_cytoband_status <-
+  consensus_seg_cytoband_status_df %>%
+  left_join(
+    consensus_seg_arm_status,
+    by = c(
+      "Kids_First_Biospecimen_ID",
+      "chromosome_arm"
+    )
+  ) %>%
+  mutate(
+    dominant_cytoband_status = case_when(
+      callable_fraction.x < (1 - uncallable_threshold) ~ "uncallable",
+      loss_fraction.x > status_threshold ~ "loss",
+      gain_fraction.x > status_threshold ~ "gain",
+      loss_fraction.x + gain_fraction.x > status_threshold ~ "unstable",
+      TRUE ~ "neutral"
+    )
+  ) %>%
+  # Filter the annotated CN data to include only neutral chromosome arms and disagreements
+  filter(
+    dominant_arm_status %in% c("neutral", "uncallable", "unstable") |
+      (
+        dominant_cytoband_status != dominant_arm_status &
+          # bands that disagree with arm, but are not neutral (or uncallable)
+          !(dominant_cytoband_status %in% c("neutral", "uncallable", "unstable"))
+      )
+  ) %>%
+  select(
+    Kids_First_Biospecimen_ID,
+    cytoband,
+    loss_fraction = loss_fraction.x,
+    gain_fraction = gain_fraction.x,
+    callable_fraction = callable_fraction.x,
+    dominant_cytoband_status
+  )
+
+# Display table
+consensus_seg_cytoband_status
+```
+
+### Determine gene-level status
+
+```{r}
+# Now create a data.frame with the gene-level status calls for the
+# neutral, uncallable, and unstable cytoband-level calls
+consensus_seg_gene_status <- consensus_seg_autosomes_df %>%
+  # Create a cytoband column that keeps only the first half of the ranges
+  # for joining purposes
+  mutate(cytoband_no_ranges = gsub("-.*", "\\1", cytoband)) %>%
+  left_join(
+    consensus_seg_cytoband_status_df,
+    by = c(
+      "biospecimen_id" = "Kids_First_Biospecimen_ID",
+      "cytoband_no_ranges" = "cytoband"
+    )
+  ) %>%
+  # Filter the annotated CN data to include only neutral cytobands and disagreements
+  filter(
+    dominant_cytoband_status %in% c("neutral", "uncallable", "unstable") |
+      (status != dominant_cytoband_status &
+         # bands that disagree with arm, but are not neutral (or uncallable)
+         !(status %in% c("neutral", "uncallable", "unstable")
```

Comment:
There are no neutral calls at the gene level, so I think we can skip this part of the filter.